skopt/optimizer/optimizer.py

```diff
@@ -467,7 +471,7 @@ def tell(self, x, y, fit=True):
                 # minimization starts from `n_restarts_optimizer` different
                 # points and the best minimum is used
                 elif self.acq_optimizer == "lbfgs":
-                    x0 = X[np.argsort(values)[:self.n_restarts_optimizer]]
+                    x0 = X[np.argsort(values)]
```

Comment:
hmm maybe something like N_random_points = 10^min(4, dimensions)? But then I guess one should make a distinction for different surrogates, as for example for tree based ones you would want to use all 10^4 as you do not apply LBFGS afterwards. 

With the second point, do you mean that you would rather repeat more often LBFGS optimization instead of random sampling? Intuitively it seems that it depends on whether acquisition function has a lot of local sub optimal minima or flat areas where solver can get stuck, and hence random sampling should help avoid such areas by providing initialization close to optimal points (but this is somewhat handwaving argument).